lib/puppet/provider/network_config/interfaces.rb

```diff
@@ -292,10 +289,24 @@ def self.format_file(filename, providers)
         stanza << "#{section} #{provider.send property}" if provider.send(property) and provider.send(property) != :absent
       end
 
+      if provider.mode == :vlan
+          if provider.options["vlan-raw-device"]
+            stanza << "vlan-raw-device #{provider.options["vlan-raw-device"]}"
+          else
+            vlan_range_regex = %r[\d{1,3}|40[0-9][0-5]]
```

Comment:
@rnelson0 well my fork is quite old and it needs a rebase against current master to be useful at all. If thwere's real interesting in accepting this feature and merging it, I could work on it